scylla-macros/src/serialize/value.rs

```diff
@@ -125,6 +137,28 @@ impl Context {
                 errors.push(err);
             }
 
+            // When name checks are skipped, fields with `allow_missing` are only
+            // permitted at the end of the struct, i.e. no field without
+            // `allow_missing` and `skip` is allowed to be after any field
+            // with `allow_missing`.
+            let invalid_default_when_missing_field = self
+                .fields
+                .iter()
+                .rev()
+                // Skip the whole suffix of <allow_missing> and <skip>.
+                .skip_while(|field| !field.attrs.skip)
+                // skip_while finished either because the iterator is empty or it found a field without both <allow_missing> and <skip>.
+                // In either case, there aren't allowed to be any more fields with `allow_missing`.
```

Comment:
Is this true? I see that we only check for `skip` and not `allow_missing` (i.e. `fields.attr.ignore_missing`).

Reply:
Is there a test for this?

Reply:
Also, I see the following in `derive_serialize_value`:
```rust
        // Filter the fields now instead of at the places that use them later
        // as it's less error prone - we just filter in one place instead of N places.
        .filter(|f| f.as_ref().map(|f| !f.attrs.skip).unwrap_or(true))
```

Doesn't this mean that there will be no fields with `skip = true` in your validation code?

Reply:
Eh, it's again the same issue as once in the past in #1024...
The `compile_fail` tests fail to compile due to another reason and thus pass, even though the check is ill-formed, as you cleverly noticed.

Reply:
Actually, the condition in the current state is too rigid. It rejects structs as e.g.:
```rust
struct Udt {
  a: i32,
  #[allow_missing]
  b: i32,
  #[allow_missing]
  c: i32,
}
```
because `c` doesn't have `#[skip]`.

Reply:
I fixed the bug. Thank you!